### PR TITLE
Port polish items from old HITL study branch

### PR DESCRIPTION
**Pull request by rpartsey**
## Motivation and Context

Port polish items from old HITL study branch. See [this](https://app.asana.com/0/home/1205684818278375/1205855517284566) Asana task (and comments) for more context.

## How Has This Been Tested
Running sandbox app locally on MacBook Pro 2021. 

Assets and habitat-sim setup the same as in [this](https://docs.google.com/document/d/1cvKuXXE2cKchi-C_O7GGVFZ5x0QU7J9gHTIETzpVKJU/edit#bookmark=id.ert94sc10d2q) SIRo Sandbox snapshot.

Launch command (--show-tutorial flag added):
```
"program": "examples/siro_sandbox/sandbox_app.py",
"args": [
    "--app-state", "socialnav_study",
    "--display-font-size", "14",
    "--disable-inverse-kinematics",
    "--never-end",
    "--first-person-mode",
    "--hide-humanoid-in-gui",
    "--width", "1300",
    "--height", "1000",
    "--gui-controlled-agent-index", "1",
    "--ang-speed", "300",
    "--save-episode-record",
    "--save-filepath-base", "my_session",
    "--show-tutorial",
    "--cfg", "examples/siro_sandbox/configs/socialnav_study.yaml",
    "--cfg-opts",
    "~habitat.task.measurements.agent_blame_measure",
    "habitat.dataset.data_path=data/datasets/floorplanner/rearrange/scratch/test/largetest_10s_500epi_2obj.json.gz",
    "habitat.simulator.agents.agent_1.articulated_agent_urdf='data/humanoids/humanoid_data/female_3/female_3.urdf'"
],

```

Note, this is not tested with rearrange app state, since it is not working on demo branch.

## Types of changes

- **Development** PR into `eundersander/SIRo_hitl_Sep5_demo_refactored` (demo branch).

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes if required.

### CODE REVIEW

File: examples/siro_sandbox/sandbox_app.py [hunk 1]
```diff
@@ -710,6 +718,12 @@ def _parse_debug_third_person(args, framebuffer_size):
         default=False,
         help="Shows an intro sequence that helps familiarize the user to the scene and task in a HITL context.",
     )
+    parser.add_argument(
```
Comment (eundersander) [inline examples/siro_sandbox/sandbox_app.py:721]: I asked @rutadesai on Slack about if we can just make this the only option for the tutorial (and thus we don't need to complicate the code with multiple tutorial-related command-line args).

File: habitat-lab/habitat/tasks/rearrange/actions/actions.py [hunk 2]
```diff
@@ -409,9 +409,12 @@ def update_base(self, fix_leg=False):
         end_pos = self._sim.step_filter(
             rigid_state.translation, target_rigid_state.translation
         )
-        # Offset the base if the base height is different between end_pos
-        # and the current state
-        if end_pos[1] != rigid_state.translation[1]:
+
+        # try_step may fail, in which case it simply returns the start argument
+        did_try_step_fail = end_pos == rigid_state.translation
```
Comment (eundersander) [inline habitat-lab/habitat/tasks/rearrange/actions/actions.py:414]: I guess this is fine. But going forward, let's keep our PRs into the HITL codebase separate from our PRs into Habitat-lab/baselines.

**Review by eundersander — COMMENTED** [review]
Good progress! Let's see if we can simplify tutorial command line args, see comment below. Also, is there a plan to address #1 and #3 from [the Asana item](https://app.asana.com/0/1205829436198608/1205855517284566), perhaps a separate PR?

File: examples/siro_sandbox/sandbox_app.py (hunk 1 above)
Comment (rpartsey) [reply examples/siro_sandbox/sandbox_app.py]: Done

**Review by eundersander — APPROVED** [review]
LGTM!

**Pull request merged**